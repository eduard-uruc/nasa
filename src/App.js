import { BrowserRouter, Routes, Route } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import "./App.css";

import Navigation from "./app/navigation";
import { Search } from "./features/search/Search";
import { Show } from "./features/show/Show";

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Navigation />}>
            <Route index element={<Search />} />
            <Route path="/show/:nasa_id" element={<Show />} />
          </Route>
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
