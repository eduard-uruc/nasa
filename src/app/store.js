import { configureStore } from "@reduxjs/toolkit";
import searchReducer from "../features/search/searchSlice";
import showReducer from "../features/show/showSlice";

export default configureStore({
  reducer: { collection: searchReducer, searchedItem: showReducer },
});
