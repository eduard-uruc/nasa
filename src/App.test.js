import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "./app/store";

import App from "./App";

describe("App component", () => {
  it("renders correctly", () => {
    const view = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});
