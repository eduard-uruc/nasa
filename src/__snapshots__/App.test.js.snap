// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <nav
          class="navbar navbar-expand-lg navbar-light bg-light"
        >
          <div
            class="container"
          >
            <span
              class="navbar-brand"
            >
              Nasa
            </span>
            <button
              aria-controls="basic-navbar-nav"
              aria-label="Toggle navigation"
              class="navbar-toggler collapsed"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="navbar-collapse collapse"
              id="basic-navbar-nav"
            >
              <div
                class="me-auto navbar-nav"
              >
                <a
                  href="/"
                  style="text-decoration: none; color: rgb(0, 0, 0); margin-top: 3px;"
                >
                  Search Page
                </a>
              </div>
            </div>
          </div>
        </nav>
        <section
          class="page-section"
        >
          <h2
            class="basic-title"
          >
            Search Page
          </h2>
          <form
            class=""
            style="margin-top: 3em;"
          >
            <div
              class="mb-3"
            >
              <input
                class="form-control"
                id="searchQuery"
                placeholder="Type a keyword"
                type="email"
                value=""
              />
              <small
                class="err-msg form-text"
              />
            </div>
            <div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="yearStart"
                    >
                      From
                    </label>
                    <input
                      class="form-control"
                      id="yearStart"
                      placeholder="Date start"
                      type="date"
                      value=""
                    />
                    <small
                      class="err-msg form-text"
                    >
                       
                       
                    </small>
                  </div>
                </div>
                <div
                  class="col"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="form-label"
                      for="yearEnd"
                    >
                      To
                    </label>
                    <input
                      class="form-control"
                      id="yearEnd"
                      placeholder="Date end"
                      type="date"
                      value=""
                    />
                    <small
                      class="err-msg form-text"
                    >
                       
                    </small>
                  </div>
                </div>
              </div>
            </div>
            <button
              class="btn btn-primary"
              disabled=""
              type="button"
            >
              Submit
            </button>
          </form>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          class="container"
        >
          <span
            class="navbar-brand"
          >
            Nasa
          </span>
          <button
            aria-controls="basic-navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="basic-navbar-nav"
          >
            <div
              class="me-auto navbar-nav"
            >
              <a
                href="/"
                style="text-decoration: none; color: rgb(0, 0, 0); margin-top: 3px;"
              >
                Search Page
              </a>
            </div>
          </div>
        </div>
      </nav>
      <section
        class="page-section"
      >
        <h2
          class="basic-title"
        >
          Search Page
        </h2>
        <form
          class=""
          style="margin-top: 3em;"
        >
          <div
            class="mb-3"
          >
            <input
              class="form-control"
              id="searchQuery"
              placeholder="Type a keyword"
              type="email"
              value=""
            />
            <small
              class="err-msg form-text"
            />
          </div>
          <div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="mb-3"
                >
                  <label
                    class="form-label"
                    for="yearStart"
                  >
                    From
                  </label>
                  <input
                    class="form-control"
                    id="yearStart"
                    placeholder="Date start"
                    type="date"
                    value=""
                  />
                  <small
                    class="err-msg form-text"
                  >
                     
                     
                  </small>
                </div>
              </div>
              <div
                class="col"
              >
                <div
                  class="mb-3"
                >
                  <label
                    class="form-label"
                    for="yearEnd"
                  >
                    To
                  </label>
                  <input
                    class="form-control"
                    id="yearEnd"
                    placeholder="Date end"
                    type="date"
                    value=""
                  />
                  <small
                    class="err-msg form-text"
                  >
                     
                  </small>
                </div>
              </div>
            </div>
          </div>
          <button
            class="btn btn-primary"
            disabled=""
            type="button"
          >
            Submit
          </button>
        </form>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
